docs/Authorization/2-legged-oauth/UsingAServiceAccount.md

```diff
@@ -390,16 +390,18 @@ public class ServiceAccountSample {
 
             // Form and sign the JWT
             Map<String, Object> headerClaims = new HashMap<String, Object>();
-            headerClaims.put("kid", credentials.get("assertionHeaderData").get("kid"));
-            headerClaims.put("alg", credentials.get("assertionHeaderData").get("alg"));
+            JSONObject jsonObject = (JSONObject)credentials.get("assertionHeaderData");
+            headerClaims.put("kid", jsonObject.get("kid"));
+            headerClaims.put("alg", jsonObject.get("alg"));
 
             // jti claim is not yet used by Vendasta, and it will be ignored in the backend; regardless, here is an example of how to make and use the claim
             String jti = UUID.randomUUID().toString();
+            jsonObject = (JSONObject)credentials.get("assertionPayloadData");
 
             String token = JWT.create()
-                    .withAudience((String) credentials.get("assertionPayloadData").get("aud"))
-                    .withIssuer((String) credentials.get("assertionPayloadData").get("iss"))
-                    .withSubject((String) credentials.get("assertionPayloadData").get("sub"))
+                    .withAudience((String) jsonObject.get("aud"))
+                    .withIssuer((String) jsonObject.get("iss"))
+                    .withSubject((String) (JSONObject)credentials.get("assertionPayloadData").get("sub"))
```

Comment:
I don't understand your last commit. Instead of renaming the variables like I suggested you changed just this one line back to almost what it was before.